keras/optimizers.py

```diff
@@ -683,6 +685,10 @@ def from_config(self, config):
         raise NotImplementedError
 
 
+if K.backend() == 'mxnet':
+    SGD, Adagrad, Adadelta, Adam, RMSprop = K.get_optimizers()
```

Comment:
The optimizers as implemented in this file should work with MXNet.  Don't replace them with native classes. For instance, current optimizers are shared across TF/Theano/CNTK even though TF has native optimizers.

Reply:
I agree and ideally, it should be all Keras engine code that manages the backend. I was thinking of getting into hands of users, get early feedback, and create a Github issue to track these and fast follow up after this PR to work on removing MXNet specific things.
Again, this requires update_* mechanisms. I will go through Theano way of handling it and get it implemented for MXNet backend.